src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,184 @@
+/* global coblocksBlockData */
+
+/**
+ * External dependencies
+ */
+import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+
+/**
+ * Internal dependencies
+ */
+import Controls from './controls';
+import OpenTable from './opentable';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { compose, usePrevious } from '@wordpress/compose';
+import {
+	Placeholder,
+	Notice,
+	Button,
+	withNotices,
+	Icon,
+	FormTokenField,
+} from '@wordpress/components';
+
+const Edit = ( props ) => {
+	const [ ridField, setRidField ] = useState( [] );
+	const [ queryResults, setQueryResults ] = useState( [] );
+	const { className, attributes } = props;
+	const [ noResultsFound, setNoResultsFound ] = useState( false );
+
+	const prevIDs = usePrevious( attributes.restaurantIDs );
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = ( token ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.then( ( json ) => {
+				const results = [];
+				for ( const item in json.items ) {
+					const r = json.items[ item ];
+					const name = `${ decodeURIComponent( r.name ) },  ${ decodeURIComponent( r.addressResponse.city ) }, ${ decodeURIComponent( r.addressResponse.country ) } (${ r.rid })`;
```

Comment:
For the sake of readability and posterity, we should try to avoid the use of single-letter variable names `const r =`. Something like `itemProperties`, or `itemProps` perhaps might be more understandable at a glance?

Reply:
resolved in 897960c